src/UnityExtension/Assets/Editor/GitHub.Unity/ApplicationCache.cs

```diff
@@ -161,21 +162,27 @@ public bool ValidateData()
 
         public void InvalidateData()
         {
-            Logger.Trace("Invalidate");
-            LastUpdatedAt = DateTimeOffset.MinValue;
-            CacheInvalidated.SafeInvoke(CacheType);
+            if (!isInvalidating)
+            {
+                Logger.Trace("Invalidate");
+                isInvalidating = true;
+                LastUpdatedAt = DateTimeOffset.MinValue;
+                CacheInvalidated.SafeInvoke(CacheType);
+            }
         }
 
         protected void SaveData(DateTimeOffset now, bool isChanged)
         {
             var isInitialized = IsInitialized;
             isChanged = isChanged || !isInitialized;
 
-            InitializedAt = !isInitialized ? now : InitializedAt;
-            LastUpdatedAt = isChanged ? now : LastUpdatedAt;
+            InitializedAt = !isInitialized || InitializedAt == DateTimeOffset.MinValue ? now : InitializedAt;
+            LastUpdatedAt = isChanged || LastUpdatedAt == DateTimeOffset.MinValue ? now : LastUpdatedAt;
```

Comment:
This is the fix for the infinite invalidation loop - if the data hasn't changed, `LastUpdated` is never set back to a valid value, causing the `ValidateData` to think the cache has timed out and causing an invalidation, which comes back here and doesn't set the `LastUpdatedAt` field, etc ad nauseaum.